src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java

```diff
@@ -54,4 +56,25 @@ public InnReachTransactionsDTO getAllTransactions(Integer offset, Integer limit,
     return transactionMapper.toDTOCollection(transactions);
   }
 
+  @Override
+  @Transactional
+  public InnReachTransactionDTO updateInnReachTransaction(UUID transactionId, InnReachTransactionDTO transaction) {
+    var oldTransaction = repository.fetchOneById(transactionId).get();
+
+    var oldTrackingId = oldTransaction.getTrackingId();
+    var oldCode = oldTransaction.getCentralServerCode();
+    var oldType = oldTransaction.getType().name();
+
+    var newTrackingId = transaction.getTrackingId();
+    var newCode = transaction.getCentralServerCode();
+    var newType = transaction.getType().name();
+
+    if (oldTrackingId.equals(newTrackingId) && oldCode.equals(newCode) && oldType.equals(newType)) {
+      repository.save(transactionMapper.toEntity(transaction));
+      return transaction;
+    } else {
+      throw new IllegalArgumentException(IMMUTABLE_TRANSACTION_FIELDS_MESSAGE);
```

Comment:
Is there a requirement to throw an exception?
Couldn't the changes just be ignored?

Reply:
Do you mean that return type of this method should be void?